### ISSUE TITLE
[FEAT] Add example for blocking iteration process

### ISSUE THREAD

**Issue by RossKen** [moj-analytical-services/splink#1430]
### Is your proposal related to a problem?

Blocking is causes confusion for many users. We are in the process of adding documentation, but some more practical ways to show the interactive process when building a pipeline would be good to bring new users along the journey.

### Describe the solution you'd like

See discussion on #1389

**Comment by RobinL** [comment]
Worth noting I'm thinking about working on some code to auto suggest blocking rules (for prediction and training) based on the new more efficient count comparisons function. I.e. to some extent (hopefully) this should be easier in future

**Comment by RossKen** [comment]
Hey @RobinL, yea I knew you were looking at some auto generated rules but wasn't sure what sort of timescales that work was on so thought these additional docs were still worth considering, given the feedback we receive on blocking.

**Comment by RobinL** [comment]
Yeah, definitely worth having something, maybe just fairly brief at the moment.  In terms of timescales, not sure,  but now the groundwork is in place I don't think it will take too long to get something basic up and running.
